File: work/pmod_swt_8ld/pmod_swt_8ld.c
#include <stdio.h>
#include <stdlib.h>
#include "/home/mi005558/pi/work/common/common.h"
#include "/home/mi005558/pi/work/gpio/gpio.h"

#define COUNT 8

int main() {
    char *SWT = JB; // Tableau des broches des interrupteurs
    char *LED = JC; // Tableau des broches des LEDs

    // Configurer les interrupteurs en entrée et les LEDs en sortie
    for (int i = 0; i < COUNT; i++) {
        gpio_set_direction(SWT[i], 0); // Configurer interrupteurs en entrée
        gpio_set_direction(LED[i], 1); // Configurer LEDs en sortie
    }

    while (1) {
        for (int i = 0; i < COUNT; i++) {
            int state = gpio_get_value(SWT[i]); // Lire l'état des interrupteurs
            if (state != -1) { // Vérification d'erreur
                gpio_set_value(LED[i], state); // Afficher sur les LEDs
            }
        }
    }

    return 0;
}

File: work/pmod_swt_8ld/pmod_swt_8ld.py
import time
import sys
from gpio import JB, JC, gpio_set_direction, gpio_get_value, gpio_set_value

# Configuration des broches en entrée (interrupteurs) et en sortie (LEDs)
for i in range(8):
    gpio_set_direction(JB[i], 0)  # Configurer les interrupteurs en entrée
    gpio_set_direction(JC[i], 1)  # Configurer les LEDs en sortie

while True:
    val = 0
    for i in range(8):
        bit = gpio_get_value(JB[i])  # Lire l'état des interrupteurs
        gpio_set_value(JC[i], bit)   # Mettre à jour les LEDs en fonction des interrupteurs
        val += bit * (1 << i)
    
    print(f"Valeur lue : {val:08b}")  # Affiche la valeur binaire lue
    time.sleep(0.5)  # Pause pour éviter une lecture trop rapide
